PedroChaparro/loomies-backend
# Feat: Generate and Claim Gyms rewards

## Includes: 

- [x] Insert loomballs in the database setup. 
- [x] Add gym_rewards fields in the gyms collection.  
- [x] Add a probability to each object and loomball to become a gym reward for common users. 
- [x] Add a probability to each object and loomball to become a gym reward for gyms owners. 
- [x] Script to remove current rewards and generate new ones. (This script should be runned as a cronjob). 
- [x] Endpoint to claim the current rewards of some gym.  

## How can I test it?

1. Drop all collections from the mongo database (Except the users one). This is not mandatory but it will prevent duplicated and outdated entries. 
2. Go to the `algorithms/database` folder and re-generate the database.
3. From the same folder, run the `update:rewards` command, that will generate initial rewards for all the gyms. 
4. Send a POST request to the new `/gyms/claim_reward` endpoint, it should: 

- Add the user id to the `rewards_claimed_by` array in the claimed gym. 
![image](https://user-images.githubusercontent.com/62714297/223524542-abe2eb4e-8c31-4de1-a76b-3a8212033bff.png)

- Add the items to the `inventory` field in the users collection.
![image](https://user-images.githubusercontent.com/62714297/223526056-e355b3f5-23da-455c-8cf5-31b8d8c2e5a8.png)
